### PR TITLE
Crud

### PR DESCRIPTION
**Pull request by yosefHesham**
### Data Processing and Models
---

- Add Model for likes
- Add Model for users
- Add Model For comments.
- Add Model for posts

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+## class User
+class User < ApplicationRecord
+  has_many :posts, foreign_key: 'user_id'
+  has_many :comments, foreign_key: 'user_id'
+  has_many :likes, foreign_key: 'user_id'
```
Comment (AnselemOdims) [inline app/models/user.rb]: - [ ] Good job on this so far, but if we reference the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) you would notice that the user foreign key is referenced with author. So please kindly change the `user_id` to `author_id`

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  has_many :comment
+  has_many :like
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  def update_posts_counter
+    author.increment!(:posts_count)
+  end
```
Comment (AnselemOdims) [inline app/models/post.rb:14]: - [ ] [OPTIONAL] Good job on this method, however we can go further by including a [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save`[Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call this method anytime a post is saved to the database.

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+## class Like
+class Like < ApplicationRecord
+  belongs_to :user
```
Comment (AnselemOdims) [inline app/models/like.rb]: - [x] Please kindly ensure that this references the user class.

Hint:
`belongs_to :author, class_name: 'User'`

File: app/models/comment.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# class Comment
+class Comment < ApplicationRecord
+  belongs_to :user
```
Comment (AnselemOdims) [inline app/models/comment.rb]: - [x] Please kindly reference the above suggestion.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @yosefHesham , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Lizdev-05 — APPROVED** [review]
Hi @yosefHesham, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- [ ] PR has a title and summary. ✔️
- [ ] No linter errors ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**